ktor-network/jvm/src/io/ktor/network/selector/SelectorManagerSupport.kt

```diff
@@ -38,7 +38,8 @@ abstract class SelectorManagerSupport internal constructor() : SelectorManager {
 //            val c = base.tracked()  // useful for debugging
 
             c.invokeOnCancellation {
-                selectable.dispose()
+                // TODO: We've got a race here (and exception erasure)!
```

Comment:
Well, who does it then? Doesn't the change lead to a registration leak? Or it will only delay ?

Reply:
It's done in `NIOSocketImpl` in `checkChannels` method. If we do it here we loose type of exception (like `SocketTimeoutException`) and `dispose()` with just a cancellation.

Reply:
So, it looks like it could be only cancelled if a read/write job is cancelled. In this case, checkChannels is invoked. So let's assume it is resolved.